### PR TITLE
[Docs] Updates to CONTRIBUTING.md 

### LINKED ISSUE #1872: Update CONTRIBUTING.md to remove the references to Bitcoin Core and produce an actual contribution guideline document
### Describe the issue

The document outlining contribution guidelines for this project appears to still contain significant references to the Bitcoin Core project it was forked from. This would probably need to be addressed and an actual document specific to the Dogecoin Project produced with input from all current contributors.

### PR DESCRIPTION
**Pull request by kregerl**
Replaces all occurrences of "bitcoin" with "dogecoin" in CONTRIBUTING.md.
Fixes #1872

### CODE REVIEW

**Comment by rnicoll** [comment]
The work in progress branch is `1.14.4-dev`, which I've switched the base for this over to. Can you rebase the PR please, I think some of these (potentially all) are fixed already.

Thanks

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ To contribute a patch, the workflow is as follows:
 
   - Fork repository
   - Create topic branch
-  - Commit patches
+  - Commit patcheshttps://git-scm.com/docs/git-rebase
```
Comment (jtoomim) [inline CONTRIBUTING.md:27]: This doesn't look right.

**Comment by ch3ck** [comment]
Apparently I repeated this work #1897 , closing my PR

**Comment by patricklodder** [comment]
This has been fixed with #2111

**Pull request closed without merging**